### DISCUSSION TITLE
Python predicted energe is different from the energy obtained by the test model

### DISCUSSION THREAD

**Discussion by lammps-vasp** [deepmodeling/deepmd-kit#1092] [Q&A]
Extract the position information below each column of POSITION TOTAL-FORCE from OUTCAR to generate POSCAR. POSCAR is shown in Figure 1 below. Use the trained model to predict the energy. It turns out that the energy predicted by python is compared with the model after the model is trained. There is a difference in the energy of the test. The test command is dp test -m frozen_model.pb -s ./data -n 196 -d result. According to the same model and the same position information, the calculated energy should be the same. What is the reason? The energy predicted by python is shown in Figure 2; the energy obtained by the training model test is Figure 3. The installed dpdata version is 0.1.19, and the deepmd version isdeepmd-kit-1.3.1-cpu-Linux-x86_64.sh.
Looking forward to replying!
![image](https://user-images.githubusercontent.com/73742098/131982610-11cd1927-e6c3-4c46-b6ba-5f6a6eb9d333.png)
POSCAR  figure1
![image](https://user-images.githubusercontent.com/73742098/131981808-f5bda2a4-3567-4f0f-a8c8-8e8cb889dd3a.png)
python predict energe figure2
![image](https://user-images.githubusercontent.com/73742098/131982004-d16ec610-eede-45b9-b616-16356faa68b4.png)
test model energe figure3

**Comment by njzjz** [comment]
What's your Python script? The order of frames has been likely to be shuffled?

**Comment by lammps-vasp** [comment]
------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "deepmodeling/deepmd-kit"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年9月4日(星期六) 上午10:54
***@***.***&gt;;
抄送:&nbsp;"Zhifeng ***@***.******@***.***&gt;;
主题:&nbsp;Re: [deepmodeling/deepmd-kit] Python predicted energe is different from the energy obtained by the test model (#1092)





 
What's your Python script? The order of frames has been likely to be shuffled?
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by lammps-vasp** [reply]
I think the structure is not disrupted, because the whole is batch operation this is python script
![XDR8(P1I8Y37V2J@1HHQO13](https://user-images.githubusercontent.com/73742098/132191790-2659d835-9d1c-48fb-bc6c-895f95bf93f8.png)

**Reply by njzjz** [reply]
Can you directly load data directory? Or you need to check if coordinates in two directories are the same.

**Reply by lammps-vasp** [reply]
Can be loaded, traverse the loop in the directory, after checking the coordinates are also one-to-one correspondence
